phone/src/apps/twitter/components/AddTweetModal.tsx

```diff
@@ -98,7 +102,16 @@ export const AddTweetModal = () => {
       images:
         images && images.length > 0 ? images.map((image) => image.link).join(IMAGE_DELIMITER) : '',
     };
-    Nui.send(TwitterEvents.CREATE_TWEET, data);
+
+    fetchNui<ServerPromiseResp<void>>(TwitterEvents.CREATE_TWEET, data).then((resp) => {
+      if (resp.status !== 'ok') {
+        return addAlert({
+          type: 'error',
+          message: t('APPS_TWITTER_CREATE_FAILED'),
+        });
+      }
+    });
+
```

Comment:
Be mindful that spamming enter here would result in multiple promise events being fired, if the previous promise isn't yet resolved.